mvlearn/embed/mcca.py

```diff
@@ -0,0 +1,590 @@
+
+    n_views_ : int
+        The number of views
+
+    n_features_ : list
+        The number of features in each fitted view
+
+    n_components_ : int
+        The number of components in each transformed view
+
+    See also
+    --------
+    KMCCA
+
+    References
+    ----------
+    .. [#1mcca] Kettenring, J. R., "Canonical Analysis of Several Sets of
+                Variables." Biometrika, 58 (1971), pp. 433-451
+    .. [#2mcca] Tenenhaus, A., et al. "Regularized generalized canonical
+                correlation analysis." Psychometrika, 76(2):257.
+
+    Examples
+    --------
+    >>> from mvlearn.embed import MCCA
+    >>> X1 = [[0., 0., 1.], [1.,0.,0.], [2.,2.,2.], [3.,5.,4.]]
+    >>> X2 = [[0.1, -0.2], [0.9, 1.1], [6.2, 5.9], [11.9, 12.3]]
+    >>> mcca = MCCA()
+    >>> mcca.fit([X1, X2])
+    MCCA()
+    >>> Xs_scores = mcca.transform([X1, X2])
+    """
+
+    def __init__(
+        self,
+        n_components=1,
+        regs=None,
+        signal_ranks=None,
+        center=True,
+        i_mcca_method="auto",
+        multiview_output=True,
+    ):
+
+        self.n_components = n_components
+        self.center = center
+        self.regs = regs
+        self.signal_ranks = signal_ranks
+        self.i_mcca_method = i_mcca_method
+        self.multiview_output = multiview_output
+
+    def _fit(self, Xs):
+        """Helper function for the `.fit` function"""
+        Xs, self.n_views_, _, self.n_features_ = check_Xs(
+            Xs, return_dimensions=True
+        )
+        centers = param_as_list(self.center, self.n_views_)
+        self.means_ = [np.mean(X, axis=0) if c else None
+                       for X, c in zip(Xs, centers)]
+        Xs = [X - m if m is not None else X for X, m in zip(Xs, self.means_)]
+
+        if self.signal_ranks is not None:
+            self.loadings_, scores, common_scores_normed, \
+                self.common_score_norms_, self.evals_ = _i_mcca(
+                    Xs,
+                    signal_ranks=self.signal_ranks,
+                    n_components=self.n_components,
+                    regs=self.regs,
+                    method=self.i_mcca_method,
+                )
+        else:
+            self.loadings_, scores, common_scores_normed, \
+                self.common_score_norms_, self.evals_ = _mcca_gevp(
+                    Xs,
+                    n_components=self.n_components,
+                    regs=self.regs
+                )
+        return scores, common_scores_normed
+
+    def inverse_transform(self, scores):
+        """
+        Transforms scores back to the original space.
+
+        Parameters
+        ----------
+        scores: array-like, shape (n_samples, n_components)
+            The CCA scores.
+
+        Returns
+        -------
+        Xs_hat : list of array-likes or numpy.ndarray
+             - Xs_hat length: n_views
+             - Xs_hat[i] shape: (n_samples, n_features_i)
+            The reconstructed original data
```

Comment:
this description is actually also not rendered well. It might be better to have it as

Xs_hat : list of array-likes or numpy.ndarray
             The reconstructed original data
             - Xs_hat length: n_views
             - Xs_hat[i] shape: (n_samples, n_features_i)

Reply:
or put an extra line between bullet points and "The reconstructed original data", like https://github.com/mvlearn/mvlearn/blob/master/mvlearn/cluster/mv_spherical_kmeans.py does in fit() for example